Doc/library/functions.rst

```diff
@@ -1720,50 +1720,84 @@ are always available.  They are listed here in alphabetical order.
    dictionary are ignored.
 
 
-.. function:: zip(*iterables)
-
-   Make an iterator that aggregates elements from each of the iterables.
-
-   Returns an iterator of tuples, where the *i*-th tuple contains
-   the *i*-th element from each of the argument sequences or iterables.  The
-   iterator stops when the shortest input iterable is exhausted. With a single
-   iterable argument, it returns an iterator of 1-tuples.  With no arguments,
-   it returns an empty iterator.  Equivalent to::
-
-        def zip(*iterables):
-            # zip('ABCD', 'xy') --> Ax By
-            sentinel = object()
-            iterators = [iter(it) for it in iterables]
-            while iterators:
-                result = []
-                for it in iterators:
-                    elem = next(it, sentinel)
-                    if elem is sentinel:
-                        return
-                    result.append(elem)
-                yield tuple(result)
-
-   The left-to-right evaluation order of the iterables is guaranteed. This
-   makes possible an idiom for clustering a data series into n-length groups
-   using ``zip(*[iter(s)]*n)``.  This repeats the *same* iterator ``n`` times
-   so that each output tuple has the result of ``n`` calls to the iterator.
-   This has the effect of dividing the input into n-length chunks.
-
-   :func:`zip` should only be used with unequal length inputs when you don't
-   care about trailing, unmatched values from the longer iterables.  If those
-   values are important, use :func:`itertools.zip_longest` instead.
-
-   :func:`zip` in conjunction with the ``*`` operator can be used to unzip a
-   list::
-
-      >>> x = [1, 2, 3]
-      >>> y = [4, 5, 6]
-      >>> zipped = zip(x, y)
-      >>> list(zipped)
-      [(1, 4), (2, 5), (3, 6)]
-      >>> x2, y2 = zip(*zip(x, y))
-      >>> x == list(x2) and y == list(y2)
-      True
+.. function:: zip(*iterables, strict=False)
+
+   Invert a series of iterables, turning rows into columns, and columns into rows.
```

Comment:
Yeah, completely agree. I think the original summary was fine:

```suggestion
   Make an iterator that aggregates elements from each of the iterables.
```

Reply:
I'll put the rows-and-columns approach out of the summary, even though I like it. 

I don't like the original summary "Make an iterator that aggregates elements from each of the iterables." The word "aggregate" is so generic, that it doesn't add any information. To aggregate can mean to add as a sum, to take an average, to do a union, [and other actions](https://docs.djangoproject.com/en/3.0/topics/db/aggregation/#cheat-sheet).

Let me think about another summary.

Reply:
Okay, I used some of Guido's words for the summary. I do think it's better than the "aggregate" sentence. I included the rows-and-columns approach outside of the summary, because I think it'll make the penny drop for many people.